Sources/PackageDescription4/PackageDependency.swift

```diff
@@ -24,6 +24,12 @@ extension Package.Dependency: Equatable {
         url: String,
         _ requirement: Package.Dependency.Requirement
     ) -> Package.Dependency {
+        // FIXME: This is suboptimal but its the only way to do this right now.
```

Comment:
Hmm, could we have made dependency into an enum:

```swift
public enum Dependency {
    case remote(RemoteDependency)
    case local(LocalDependency)
}

public class RemoteDependency {
    /// The dependency requirement.
    public enum Requirement {
        case exactItem(Version)
        case rangeItem(Range<Version>)
        case revisionItem(String)
        case branchItem(String)
    }

    /// The url of the dependency.
    public let url: String

    /// The dependency requirement.
    public let requirement: Requirement

    /// Create a dependency.
    init(url: String, requirement: Requirement) {
        self.url = url
        self.requirement = requirement
    }
}

public class LocalDependency {
    public let path: String
}
```

Reply:
Possibly...I plan to revisit these data structure as part of upcoming runtime refactor.

Reply:
Also note that we can't change the public facing data structures.

Reply:
Can’t we change the public facing data structures as part of a 4.2 PD?